### ISSUE TITLE
zip and zipr fail to use non-ASCII paths

### ISSUE THREAD

**Issue by FrancoisR95** [r-lib/zip#53]
Both zipr and zip fail to use non-ASCII paths, at least on Windows. I open this issue as a more general bug than https://github.com/r-lib/zip/issues/46 (which refers to the consequence with openxlsx) but it's basically the same bug.

Here is a **code for reproducing this issue**:
<pre><code>library(zip)

# --- Create 1 test directory and 2 sub-directories with accent, with 1 file inside the second one ----
dir.create("C:/Test/")
dir.create("C:/Test/compressed_é")
dir.create(dir_with_accent <- "C:/Test/uncompressed_é")
cat("this is one file", file = file.path(dir_with_accent, "file1.txt"))

# --- Create an empty zip file with accents in its name and path ----
# The empty zip file is made with the following 22 bytes : PK\x05\x06\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00
empty_zip_data <- as.raw(c(0x50, 0x4B, 0x05, 0x06, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00))
zipfile_with_accent <- "C:/Test/compressed_é/compressé.zip"
writeBin(empty_zip_data, zipfile_with_accent)

# --- Using zip & zipr with zipfile_with_accent & dir_with_accent (fails for both zip and zipr) ----
zipr(zipfile_with_accent, dir_with_accent)
zip(zipfile_with_accent, dir_with_accent)
</code></pre>
The execution gives the following errors:
<pre>> zipr(zipfile_with_accent, dir_with_accent)
Error in zip_internal(zipfile, files, recurse, compression_level, append = FALSE,  : 
  zip error: `Cannot add file `uncompressed_Ã©/file1.txt` to archive `C:/Test/compressed_é/compressé.zip`` in file `zip.c:394`
> zip(zipfile_with_accent, dir_with_accent)
Error in zip_internal(zipfile, files, recurse, compression_level, append = FALSE,  : 
  zip error: `Cannot add directory `C:/Test/uncompressed_é/` to archive `C:/Test/compressed_é/compressé.zip`` in file `zip.c:387`
</pre>
But if you remove the accents in the names and path, then there is no error, at least with zipr.
With zip, I still get this error even when there are no accents, but I suppose this is because there is a non-ASCII character in my Windows username (which is "François").

**Comment by vituri** [comment]
I have the same problem in Windows, with any special character like á, ã, ç, ó, etc. The only workaround was to remove those characters.

**Comment by FrancoisR95** [comment]
@vituri It's nice than you have this workaround. People like me who have a special character in their Windows username cannot use  such workaround in most cases. This makes several packages (such as openxlsx 4.1.4) unusable on my system because they rely on this buggy zip package. Let's hope this bug will be fixed soon.

**Comment by gaborcsardi** [comment]
@FrancoisR95 I'll try to fix this soon. 

In general, R often has problems with non-ascii path, and one way to work around this is to define a new drive for your home directory (e.g. `H:`), and use that in R.

**Comment by FrancoisR95** [comment]
@gaborcsardi Many thanks.

**(1) Directories (I changed the tmp path in order to remove the non-ASCII character)**
My R home directory is very standard : "C:/Program Files/R/R-4.0.2/"
Could the space in "Program Files" be a problem, although this is just the default Windows installation for anyone?
My R tmp directory contained a non-ASCII character, so I have now changed it (by creating in the R home directory a Renviron.site file containing the following line: <code>TMPDIR=C:/R-home/rtmp</code>).
Now, all the following directories do not contain any special character (other than the space in R home as mentioned above):
````R
> R.home()
[1] "C:/PROGRA~1/R/R-40~1.2"
> getwd()
[1] "C:/R-home"
> tempdir()
[1] "C:/R-home/rtmp\\Rtmpa0TIPC"
> .libPaths()
[1] "C:/R-home/Rpackages/win"            "C:/Program Files/R/R-4.0.2/library"
````
**(2) Still a bug in zip**
However, if I run the following code, which do not contain non-ASCII character:
````R
library(zip)

# --- Create 1 test directory and 2 sub-directories without accent, with 1 file inside the second one ----
dir.create("C:/Test/")
dir.create("C:/Test/compressed")
dir.create(dir_without_accent <- "C:/Test/uncompressed")
cat("this is one file", file = file.path(dir_without_accent, "file1.txt"))

# --- Create an empty zip file without accent  ----
# The empty zip file is made with the following 22 bytes : PK\x05\x06\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00
empty_zip_data <- as.raw(c(0x50, 0x4B, 0x05, 0x06, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00))
zipfile_without_accent <- "C:/Test/compressed/compresse.zip"
writeBin(empty_zip_data, zipfile_without_accent)

# --- Using zip & zipr with zipfile_without_accent & dir_without_accent ----
zipr(zipfile_without_accent, dir_without_accent)
zip(zipfile_without_accent, dir_without_accent)
```` 
I still get an error (with zip but not with zipr): 
````R
Error in zip_internal(zipfile, files, recurse, compression_level, append = FALSE,  : 
  zip error: `Cannot add directory `C:/Test/uncompressed/` to archive `C:/Test/compressed/compresse.zip`` in file `zip.c:387`
```` 
**(3) But no more zip bug with openxlsx 4.1.5**
The good news is that after the change of tmp directory as described above (1), I upgraded to openxlsx from 4.1.3 to 4.1.5, and the previous zip bug (https://github.com/r-lib/zip/issues/46) was not seen anymore (it was previously happening in openxlsx 4.1.4 & 4.1.5). I had also upgraded R from 4.0.0 to 4.0.2 yesterday, which may (or not) have had an impact.

**Comment by gaborcsardi** [comment]
> I still get an error (with zip but not with zipr):

That's because you cannot store full windows paths in a zip file. I.e. `c:\` and `zip()` (by default) takes the path as is.

**Comment by gaborcsardi** [comment]
Unicode paths are fixed by #59. I'll think about the `c:` prefix, maybe we should (optionally?) drop it. But in the first place you should not create zip files with absolute paths.

**Comment by FrancoisR95** [comment]
Hello, I confirm that with the current version of zip (2.1.1), I don't get anymore the error "in file `zip.c:394`" with zipr which I had described when opening this issue, even when using a non-ACSII temporary directory.
Many thanks @gaborcsardi for the correction!
